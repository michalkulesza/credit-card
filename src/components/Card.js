import React from "react";
import "./Card.scss";

import Chip from "../res/chip.png";
import Amex from "../res/amex.png";

const Card = ({ ccNum, name, month, year, cvvActive }) => {
  let cardNum = [];
  const defCardNum = "0000 0000 0000 0000";
  let defName = "FULL NAME";

  if (ccNum.length === 0) {
    for (let i = 0; i < defCardNum.length; i++) {
      cardNum.push(
        <div className="card-number-char" key={Math.random() * i}>
          {defCardNum[i]}
        </div>
      );
    }
  } else {
    for (let i = 0; i < ccNum.length; i++) {
      cardNum.push(
        <div className="card-number-char" key={Math.random() * i}>
          {ccNum[i]}
        </div>
      );
    }
  }

  return (
    <>
      <div className={cvvActive ? "card-wrapper card-front flipped" : "card-wrapper card-front"}>
        <div className="card-row card-row-1">
          <div className="card-hologram">
            <img src={Chip} alt="Card's chip" />
          </div>
          <div className="card-logo">
            <img src={Amex} alt="Card's type logo" />
          </div>
        </div>
        <div className="card-row card-row-2">
          <div className="card-number">{cardNum}</div>
        </div>
        <div className="card-row card-row-3">
          <div className="card-name">
            <div className="card-name-title">Card Holder</div>
            <div className="card-name-name">{name.length === 0 ? defName : name}</div>
          </div>
          <div className="card-expires">
            <div className="card-expires-title">Expires</div>
            <div className="card-expires-date">
              <div className="card-expires-date-month">{month === 0 ? "MM" : month}</div>
              <span>/</span>
              <div className="card-expires-date-year">{year === 0 ? "YY" : year.substring(2)}</div>
            </div>
          </div>
        </div>
      </div>
      <div className={cvvActive ? "card-wrapper card-back flipped" : "card-wrapper card-back"}>
        <div className="card-row card-row-1">
          <div className="card-stripe"></div>
        </div>
        <div className="card-row card-row-2">
          <div className="card-back-middle">
            <div className="card-cvv-title">CVV</div>
            <div className="card-white-stripe"></div>
          </div>
        </div>
        <div className="card-row card-row-5"></div>
      </div>
    </>
  );
};

export default Card;
