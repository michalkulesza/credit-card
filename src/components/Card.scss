.card-wrapper {
  height: 270px;
  width: 430px;
  background-color: rgb(122, 145, 158);
  margin-bottom: -135px;
  z-index: 1;
  border-radius: 15px;
  padding: 25px 25px;
  box-shadow: 0 0 5px 5px rgba($color: #000000, $alpha: 0.05);
  color: white;
}

.card-row {
  height: 33.333%;
  width: 100%;
  display: flex;

  &.card-row-1 {
    justify-content: space-between;

    .card-hologram {
      height: 50px;
    }

    .card-logo {
      max-height: 37px;
    }
  }

  &.card-row-2 {
    display: flex;
    justify-content: center;
    align-items: center;

    .card-number {
      display: flex;
      height: auto;

      .card-number-char {
        width: 20px;
        font-size: 24px;
        font-weight: 500;
      }
    }
  }

  &.card-row-3 {
    display: flex;
    justify-content: space-between;
    align-items: flex-end;

    .card-name {
      display: flex;
      flex-direction: column;
      height: auto;

      .card-name-title {
        height: 1.2em;
        font-size: 0.8em;
        font-weight: 600;
        opacity: 0.6;
      }

      .card-name-name {
        height: 1.4em;
        font-size: 1.1em;
        font-weight: 600;
      }
    }

    .card-expires {
      display: flex;
      flex-direction: column;
      height: auto;

      .card-expires-title {
        height: 1.2em;
        font-size: 0.8em;
        font-weight: 600;
        opacity: 0.6;
      }

      .card-expires-date {
        height: 1.4em;
        font-size: 1.1em;
        font-weight: 600;
        display: flex;
      }
    }
  }
}
