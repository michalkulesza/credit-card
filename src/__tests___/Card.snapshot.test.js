import React from "react";
import Card from "../components/Card";
import renderer from "react-test-renderer";

describe("Card component renders correctly with default props", () => {
  it("renders correctly", () => {
    const rendered = renderer.create(<Card />);
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
