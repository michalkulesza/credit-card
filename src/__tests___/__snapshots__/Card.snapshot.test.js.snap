// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders correctly with default props renders correctly 1`] = `
Array [
  <div
    className="card-wrapper card-front"
  >
    <div
      className="card-overlay"
    >
      <div
        className="card-row card-row-1"
      >
        <div
          className="card-hologram"
        >
          <img
            alt="Card's chip"
            src="chip.png"
          />
        </div>
        <div
          className="card-logo"
        />
      </div>
      <div
        className="card-row card-row-2"
      >
        <div
          className="card-number"
        >
          <div
            className="card-number-char"
          >
            *
          </div>
          <div
            className="card-number-char"
          >
            *
          </div>
          <div
            className="card-number-char"
          >
            *
          </div>
          <div
            className="card-number-char"
          >
            *
          </div>
          <div
            className="card-number-char"
          >
             
          </div>
          <div
            className="card-number-char"
          >
            *
          </div>
          <div
            className="card-number-char"
          >
            *
          </div>
          <div
            className="card-number-char"
          >
            *
          </div>
          <div
            className="card-number-char"
          >
            *
          </div>
          <div
            className="card-number-char"
          >
             
          </div>
          <div
            className="card-number-char"
          >
            *
          </div>
          <div
            className="card-number-char"
          >
            *
          </div>
          <div
            className="card-number-char"
          >
            *
          </div>
          <div
            className="card-number-char"
          >
            *
          </div>
          <div
            className="card-number-char"
          >
             
          </div>
          <div
            className="card-number-char"
          >
            *
          </div>
          <div
            className="card-number-char"
          >
            *
          </div>
          <div
            className="card-number-char"
          >
            *
          </div>
          <div
            className="card-number-char"
          >
            *
          </div>
        </div>
      </div>
      <div
        className="card-row card-row-3"
      >
        <div
          className="card-name"
        >
          <div
            className="card-name-title"
          >
            Card Holder
          </div>
          <div
            className="card-name-name"
          >
            FULL NAME
          </div>
        </div>
        <div
          className="card-expires"
        >
          <div
            className="card-expires-title"
          >
            Expires
          </div>
          <div
            className="card-expires-date"
          >
            <div
              className="card-expires-date-month"
            >
              MM
            </div>
            <span>
              /
            </span>
            <div
              className="card-expires-date-year"
            >
              YY
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="card-wrapper card-back"
  >
    <div
      className="card-row card-row-1"
    >
      <div
        className="card-stripe"
      />
    </div>
    <div
      className="card-row card-row-2"
    >
      <div
        className="card-back-middle"
      >
        <div
          className="card-cvv-title"
        >
          CVV
        </div>
        <div
          className="card-white-stripe"
        >
          <div
            className="card-cvv"
          />
        </div>
      </div>
    </div>
    <div
      className="card-row card-row-5"
    />
  </div>,
]
`;
